pkg/frontend/querymiddleware/stats.go

```diff
@@ -98,8 +98,12 @@ func (s queryStatsMiddleware) populateQueryDetails(ctx context.Context, req Metr
 	if details == nil {
 		return
 	}
-	details.Start = time.UnixMilli(req.GetStart())
-	details.End = time.UnixMilli(req.GetEnd())
+	if details.Start.IsZero() || details.Start.After(time.UnixMilli(req.GetStart())) {
```

Comment:
Note to reviewers: I'm making the assumption that remote read requests contain related queries so it makes sense to calculate the min of start time and max of end time from a set queries in a remote read request. I'm not sure if this is true, people can easily implement a proxy that batches quires and have nothing in common between them.

Reply:
I think that this is a fair assumption for the Prometheus use case.